keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,504 @@
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Add e2e tests for the feature (must be done before beta)
+
+#### Beta
+
+- Explicitly require hostUsers option to be `false` if this option is enabled.
+    - Otherwise, this option effectively becomes another "privileged" field
+
+#### GA
+
+- Allowing time for feedback
+
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+Turn off the feature gate to turn off the feature.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+The feature gate is only processed by the API server--Kubelet has no awareness of it. API server will scrub the ProcMount field from the request
+if it doesn't support the feature gate. Since all supported Kubelet versions support ProcMountType field, there's no version skew worry.
+API server can have the feature gate toggled without worrying about doing the same for Kubelets.
```

Comment:
This is a bit confusing, the feature gate applies to both kube-apiserver and kubelet, in which case you should answer the question what happens when you turn the feature on one, and the other has it disabled?

Reply:
the feature gate only applies to KAS, kubelet doesn't do any feature gate checks, it just accepts from KAS and processes it, regardless of whether the feature is enabled in the kubelet or not

Reply:
I'm guessing that's from a time when feature gates were not a thing. In which case I'd put an explicit information how this is currently implemented and update also `kep.yaml` since it's misleadingly pointing to both kube-apiserver and kubelet as having that feature gate.